subsys/emul/i2c/emul_sbs_gauge.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Emulator for SBS 1.1 compliant smart battery fuel gauge.
+ * This emulator is *NOT* fully featured yet, only enough to get the sbs_gauge driver tests passing.
+ */
+
+#define DT_DRV_COMPAT sbs_sbs_gauge
+
+#define LOG_LEVEL CONFIG_SENSOR_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sbs_sbs_gauge);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/emul.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/i2c_emul.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/drivers/spi_emul.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <sbs_gauge.h>
+
+/** Run-time data used by the emulator */
+struct sbs_gauge_emul_data {
+	/** I2C emulator detail */
+	struct i2c_emul emul_i2c;
+	/** sbs_gauge device being emulated */
+	const struct device *dev;
+	uint8_t pmu_status;
+	/** Current register to read (address) */
+	uint32_t cur_reg;
+};
+
+/** Static configuration for the emulator */
+struct sbs_gauge_emul_cfg {
+	/** Chip registers */
+	uint8_t *reg;
```

Comment:
Why are the registers of the chip in `_cfg` and not `_data`? And why is `reg` just a pointer and not an array inside the structure?

Reply:
Most of these actually aren't needed and won't be needed now that I think of it. Thanks.